bulk-ingester/src/test/java/org/alfresco/hxi_connector/bulk_ingester/processor/mapper/AlfrescoPropertyMapperTest.java

```diff
@@ -0,0 +1,489 @@
+    }
+
+    @Test
+    void shouldProcessSerializableProperties()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String colors = "colors";
+        List<String> colorsValue = List.of("blue", "red", "black");
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(colors, serializableValue(SerializationUtils.serialize(colorsValue)))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, colors).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(colors, property.get().getKey());
+        assertEquals(colorsValue, property.get().getValue());
+    }
+
+    @Test
+    void shouldNotProcessUnsupportedTypes()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String someRandomUnsupportedPropertyType = "someRandomUnsupportedType";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(someRandomUnsupportedPropertyType, propertyValue(PATH))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, someRandomUnsupportedPropertyType).performMapping();
+
+        // then
+        assertFalse(property.isPresent());
+    }
+
+    @Test
+    void shouldProcessContentComplexProperty()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String content = "content";
+        long contendId = 1;
+
+        ContentMetadata contentMetadata = contentMetadata(contendId);
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(content, contentValue(contendId))));
+        alfrescoNode.setContentData(
+                Set.of(contentMetadata));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, content).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(content, property.get().getKey());
+        assertEquals(contentMetadata, property.get().getValue());
+    }
+
+    @Test
+    void shouldProcessDateComplexProperty()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String createdAt = "createdAt";
+        String createdAtString = "2024-01-31T10:15:30+00:00";
+        ZonedDateTime createdAtValue = ZonedDateTime.parse(createdAtString);
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(createdAt, dateValue(createdAtString))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, createdAt).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(createdAt, property.get().getKey());
+        assertEquals(createdAtValue, property.get().getValue());
```

Comment:
and also I am wondering if we actually need it

Reply:
I think we agreed we'd just send timestamps to Hx Insight, but if we can consistently send ISO format instead then great.

Reply:
good point, I'll convert it to timestamp then

Reply:
done